toqito/matrix_ops/commutant.py

```diff
@@ -0,0 +1,67 @@
+"""Module for computing the commutant of a set of matrices."""
+
+import numpy as np
+from scipy.linalg import null_space
+
+
+def commutant(A: np.ndarray | list[np.ndarray]) -> list[np.ndarray]:
+    r"""Compute an orthonormal basis (in the Hilbert-Schmidt inner product).
+
+    For the algebra of matrices that commute with each matrix in A.
+
+
+    The commutant condition is given by:
+
+    .. math::
+        A X = X A \\quad \forall A \\in \\mathcal{A}
+
+    This can be rewritten as:
+
+    .. math::
+        (A \\otimes I - I \\otimes A^T) \text{vec}(X) = 0
+
+    where :math:`\text{vec}(X)` denotes the vectorization of :math:`X`. The null space of this
+    equation provides a basis for the commutant.
+
+    Parameters
+    ==========
+    A : np.ndarray or list[np.ndarray]
+        A single matrix or a list of square matrices of the same dimension.
+
+    Returns
+    =======
+    list[np.ndarray]
```

Comment:
You likely want to refer to how we list the parameters etc. from other functions. For instance, if you take a look at (for instance):
https://github.com/vprusso/toqito/blob/master/toqito/channel_props/choi_rank.py#L9

You'll notice how we list the parameters of the function. Also note that since we are imposing type-hinting, we don't want to duplicate this information in the docstring.